Classes/Bookmark/BookmarksViewController.swift

```diff
@@ -0,0 +1,218 @@
+        updateRightBarItem()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+    }
+    
+    // MARK: Private API
+    
+    func updateRightBarItem() {
+        navigationItem.rightBarButtonItem?.isEnabled = bookmarkStore.bookmarks.count > 0
+    }
+    
+    @objc private func onClearAll(sender: UIBarButtonItem) {
+        let alert = UIAlertController.configured(
+            title: NSLocalizedString("Bookmarks", comment: ""),
+            message: NSLocalizedString("Clear all bookmarks?", comment: ""),
+            preferredStyle: .alert
+        )
+        
+        alert.addActions([
+            AlertAction.clearAll({ [weak self] _ in
+                self?.bookmarkStore.clear()
+                self?.tableView.reloadData()
+                self?.updateRightBarItem()
+            }),
+            AlertAction.cancel()
+        ])
+        
+        present(alert, animated: true)
+    }
+    
+    func filter(query: String?) {
+        if let query = query {
+            filterdBookmarks = filtered(array: bookmarkStore.bookmarks, query: query)
+        }
+        else {
+            filterdBookmarks = nil
+        }
+        self.tableView.reloadData()
+    }
+    
+    func getBookmarks() -> [BookmarkModel] {
+        if let bookmarks = filterdBookmarks {
+            return bookmarks
+        }
+        else {
+            return bookmarkStore.bookmarks
+        }
+    }
+    
+    // MARK: TabNavRootViewControllerType
+    
+    func didSingleTapTab() {
+        
+    }
+    
+    func didDoubleTapTab() {
+        // TODO: Scroll to top for table view
+    }
+    
+    // MARK: UITableViewDataSource
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return getBookmarks().count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier)
+        
+        if cell == nil {
+            cell = UITableViewCell.init(style: .subtitle, reuseIdentifier: cellIdentifier)
+        }
+        
+        let bookmark = getBookmarks()[indexPath.row]
+        
+        let titleLabel = "\(bookmark.owner)/\(bookmark.name)"
+        cell?.textLabel?.text = bookmark.type == .repo ? titleLabel : titleLabel + " #\(bookmark.number)"
+        cell?.detailTextLabel?.text = bookmark.title
+        cell?.textLabel?.numberOfLines = 0
+        cell?.detailTextLabel?.numberOfLines = 0
+        cell?.accessibilityTraits |= UIAccessibilityTraitButton
+        cell?.isAccessibilityElement = true
+        cell?.accessibilityLabel = cell?.contentView.subviews
+            .flatMap { $0.accessibilityLabel }
+            .reduce("", { $0 + ".\n" + $1 })
+
+        cell?.imageView?.image = bookmark.getBookmarkImageIcon()
+        cell?.imageView?.tintColor = Styles.Colors.Blue.medium.color
+        return cell!
```

Comment:
This implementation feels weird to me. Especially keeping the cell optional for so long and unwrapping at the end. We should also never have to call https://github.com/rnystrom/GitHawk/pull/601/files/a6266a248f339c6820d04f39d1e474e91c09f04e#diff-1e305760accc5a494351ba0602fa475fR130 - it's handled by the system in the line above I think?

At least, I think we should fail fast if the cell is invalid, and afterwards we can then just use `cell` as a non-optional.